### ISSUE TITLE
Clients Not Connecting After Upgrade

### ISSUE THREAD

**Issue by tedtms** [lucent-sea/Remotely#141]
Jay,

I got my server updated using the process you listed in [my other ticket](https://github.com/lucent-sea/Remotely/issues/138) but unfortunately after performing it, all of my remote clients are showing offline.

I've [attached a log file](https://github.com/lucent-sea/Remotely/files/5328714/logsample.txt) that shows the error they're reporting; I'm coming from a rather old version to your latest one, is there an intermediary upgrade I should perform before jumping right to the latest stable release?

Thanks.
Ted

**Comment by lucent-sea** [comment]
There have been breaking changes along the way.  I haven't made much of an effort to make things backwards compatible or make upgrades work beyond 2 or so version jumps.  Nor have I been tracking them.

Unfortunately, there are a lot of things I don't do because it's too much effort for a free project.

I'm afraid you might need to upgrade one or two versions at a time and wait for the clients to upgrade.  Or if you have some other means of scheduling a one-time task, you can schedule a script to download and install.

- Jared

**Comment by tedtms** [comment]
Okay, I'll give that a shot -- when the server is upgraded, are the clients automatically upgraded when they connect?

**Comment by tedtms** [comment]
So I did the upgrade, and even started with a fresh database, but now any clients I install are unable to connect to the server after completing the installation. They download the installer files fine, but after starting the service I get these errors in the logs, even after a reboot. I've also tested it on multiple machines.

`2020-10-07 08:38:01.177	[Info]	Service Updater: Version is current.
2020-10-07 08:38:17.546	[Warning]	Failed to connect to server.  Internet connection may be unavailable.
2020-10-07 08:38:17.548	[Warning]	The server disconnected before the handshake could be started.	   at Microsoft.AspNetCore.SignalR.Client.HubConnection.HandshakeAsync(ConnectionState startingConnectionState, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncCore(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncCore(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncInner(CancellationToken cancellationToken)
   at System.Threading.Tasks.ForceAsyncAwaiter.GetResult()
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsync(CancellationToken cancellationToken)
   at Remotely.Agent.Services.AgentSocket.Connect() in C:\ProgramData\BuildAgent\_work\1\s\Agent\Services\AgentSocket.cs:line 61	Microsoft.AspNetCore.SignalR.Client.Core
2020-10-07 08:38:17.655	[Info]	Websocket closed.  Reconnecting in 28 seconds...`

Is this something you've seen before?
